src/imp/openssl.rs

```diff
@@ -265,7 +282,10 @@ impl TlsConnector {
         if let Some(ref identity) = builder.identity {
             connector.set_certificate(&identity.0.cert)?;
             connector.set_private_key(&identity.0.pkey)?;
-            for cert in identity.0.chain.iter().rev() {
+            for cert in identity.0.chain.iter() {
```

Comment:
I do not believe this change is correct: https://github.com/sfackler/rust-native-tls/commit/05fb5e583be589ab63d9f83d986d095639f8ec44

Reply:
> When sending a certificate chain, extra chain certificates are sent in order following the end entity certificate.
This is from https://www.openssl.org/docs/manmaster/man3/SSL_CTX_add_extra_chain_cert.html , though looking at https://www.openssl.org/docs/manmaster/man3/PKCS12_parse.html I don't see any documentation about ordering restrictions for `PKCS12_parse`.

Reply:
There really needs to be a cross-platform test for this behavior.